lib/ember-addon-main.js

```diff
@@ -57,6 +57,11 @@ module.exports = {
     return this._cachedShouldColocateTemplates;
   },
 
+  getTemplateCompiler(inputTree, htmlbarsOptions) {
+    const TemplateCompiler = require('./template-compiler-plugin');
+    return new TemplateCompiler(inputTree, htmlbarsOptions);
+  },
```

Comment:
I'd much prefer to provide a `transpileTree` API akin to the one in ember-cli-babel (see [documentation here](https://github.com/babel/ember-cli-babel#additional-trees)) which could be used from your own addon.

In that setup, you would unregister `ember-cli-htmlbars` and register your own addon as the preprocessor. Telling folks to monkey patch our internal methods seems pretty error prone to me (both for us, and for them) 🤔.

Reply:
I've renamed the method to `transpileTree` and removed the README instructions telling people they should monkey patch this addon.